docs/devcontainerjson-reference.md

```diff
@@ -0,0 +1,172 @@
+| `updateContentCommand` | string,<br>array  | This command is the second of three that finalizes container setup when a dev container is created. It executes inside the container after `onCreateCommand` whenever new content is available in the source tree during the creation process.<br><br />It will execute at least once, but GitHub Codespaces will also periodically execute the command to refresh available prebuilt codespaces. Like `onCreateCommand` in GitHub Codespaces, it can only take advantage of repository and org scoped secrets or permissions. |
+| `postCreateCommand` | string,<br>array | This command is the last of three that finalizes container setup when a dev container is created. It happens after `updateContentCommand` and once the dev container has been assigned to a user for the first time.<br><br />In GitHub Codespaces, this command can always take advantage of user specific secrets and permissions. |
+| `postStartCommand` | string,<br>array | A command to run each time the container is successfully started. |
+| `postAttachCommand` | string,<br>array | A command to run each time a tool has successfully attached to the container. |
+| `waitFor` | enum | An enum that specifies the command any tool should wait for before connecting. Defaults to `updateContentCommand`. This allows you to use `onCreateCommand` or `updateContentCommand` for steps that must happen before `devcontainer.json` supporting tools connect while still using `postCreateCommand` for steps that can happen behind the scenes afterwards. |
+
+For each command property, if the value is a single string, it will be run in `/bin/sh`. Use `&&` in a string to execute multiple commands. For example, `"yarn install"` or `"apt-get update && apt-get install -y curl"`. The array syntax `["yarn", "install"]` will invoke the command (in this case `yarn`) directly without using a shell. Each fires after your source code has been mounted, so you can also run shell scripts from your source tree. For example: `bash scripts/install-dev-tools.sh`
+
+If one of the lifecycle scripts fails, any subsequent scripts will not be executed. For instance, if `postCreateCommand` fails, `postStartCommand` and any following scripts will be skipped.
+
+## Minimum host requirements
+
+While `devcontainer.json` does not focus on hardware or VM provisioning, it can be useful to know your container's minimum RAM, CPU, and storage requirements. This is what the `hostRequirements` properties allow you to do. Cloud services like GitHub Codespaces use these properties to automatically default to the best compute option available, while in other cases, you will be presented with a warning if the requirements are not met.
```

Comment:
Similar comment here. Rather than "Cloud services like GitHub Codespaces use", we could say "Cloud Services can use"

Reply:
Good catch, updated!